generate-err-logs.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+output_file="deployment-err.log"
+> $output_file
+
+namespaces=$(kubectl get namespaces -o=jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')
+
+for namespace in $namespaces; do
+    events=$(kubectl get events -n $namespace --field-selector type=Warning -o jsonpath='{range .items[*]}{.involvedObject.kind}{" "}{.involvedObject.name}{" "}{.message}{" ("}{.reason}{")\n"}{end}')
+
+    echo "$events" | while read -r kind name message; do
+        if [ "$kind" == "Pod" ]; then
+            pod=$name
+            pod_logs=$(kubectl logs -n $namespace $pod --all-containers=true 2>&1)
+            pod_definition=$(kubectl get pod -n $namespace $pod -o yaml 2>&1)
+            cat <<EOF | tee -a $output_file
+===== Logs for pod: $pod in namespace: $namespace =====
+$pod_logs
+
+===== Definition for pod: $pod in namespace: $namespace =====
+$pod_definition
+
+===== Event message: $message =====
+
+
+EOF
+        fi
+    done
+done
```

Comment:
Can we add a main function like we have in the other scripts?

Reply:
Added